pinc/Settings.inc

```diff
@@ -10,12 +10,13 @@ class Settings
     // Multi-dimensional array containing all user settings. Each setting
     // may have one or more value associated with it. Format is:
     //     $settings_array[$setting] = array($value1, ...);
-    private $settings_array = [];
+    /** @var array<string, mixed[]> */
+    private array $settings_array = [];
 
     // username of subject user.
-    private $username = null;
+    private ?string $username = null;
 
-    public function __construct($name)
+    public function __construct(?string $name)
```

Comment:
The comment below this says that `$name` could be an integer or a boolean too (which honestly I think is very unlikely). If we feel confident this type annotation we should update the comment below.

Reply:
Good catch, I also feel fairly confident about this type so amended the comment below.